lib/plugins/aws/provider.js

```diff
@@ -2179,7 +2179,16 @@ Object.defineProperties(
         }
 
         const [repositoryName, imageTag] = image.slice(image.indexOf('/') + 1).split(':');
-        const registryId = image.split('.')[0];
+        const parts = image.split('.');
+        const registryId = parts[0];
+        const region = parts[3];
+        const serviceRegion = this.getRegion();
+        if (region !== serviceRegion) {
```

Comment:
It is probably worth checking the partition matches, too. ;)

Reply:
Could you be more specific ? I assumed that the format was `https://{aws_account_id}.dkr.ecr.{region}.amazonaws.com.`  where only region and aws_account_id are mutable values according to [https://docs.aws.amazon.com/AmazonECR/latest/userguide/Registries.html](https://docs.aws.amazon.com/AmazonECR/latest/userguide/Registries.html)

Only arn's have partitions right ?

Reply:
Oh, if we already enforce `.amazonaws.com.`, then we're fixing the partition.

Reply:
I'm unsure. I assumed the format was already being validated. Though this change isn't really format validation.